frontend/src/metabase/query_builder/containers/use-create-question.ts

```diff
@@ -0,0 +1,36 @@
+import { useCallback } from "react";
+
+import type { ScheduleCallback } from "metabase/hooks/use-callback-effect";
+import { useDispatch } from "metabase/lib/redux";
+import {
+  apiCreateQuestion,
+  setUIControls,
+  updateUrl,
+} from "metabase/query_builder/actions";
+import type Question from "metabase-lib/v1/Question";
+
+interface UseCreateQuestionParams {
+  scheduleCallback?: ScheduleCallback;
+}
+
+export const useCreateQuestion = ({
+  scheduleCallback,
+}: UseCreateQuestionParams) => {
+  const dispatch = useDispatch();
+
+  return useCallback(
+    async (newQuestion: Question) => {
+      const shouldBePinned =
+        newQuestion.type() === "model" || newQuestion.type() === "metric";
+      const createdQuestion = await dispatch(
+        apiCreateQuestion(newQuestion.setPinned(shouldBePinned)),
```

Comment:
Nice catch! I see in the original code we didn't have `await dispatch(...)` but this is indeed a Thunk.